src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,445 @@
+    private static final String MESSAGE_FORMAT_DELETE_USAGE = "Usage: %s <task number>";
+
+    private static final String MESSAGE_FORMAT_TASK_ALREADY_MARKED = "Task #%d is already marked as done.";
+    private static final String MESSAGE_FORMAT_TASK_MARKED = "Task marked as done:\n  %s";
+    private static final String MESSAGE_FORMAT_TASK_ADDED = "Got it. Task added:\n  %s\nThere are %d tasks in the list.";
+    private static final String MESSAGE_FORMAT_EXCEPTION = "An exception has occurred.\n%s";
+    private static final String MESSAGE_FORMAT_TASK_DELETED = "Task deleted:\n  %s\nThere are %d tasks left in the list.";
+    private static final String MESSAGE_FORMAT_CREATE_FILE_FAILED = "Fail to create file - %s";
+    private static final String MESSAGE_FORMAT_FILE_NOT_FOUND = "File not found - %s";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
+
+    /**
+     * Prints message within horizontal lines.
+     *
+     * @param message The message to print.
+     */
+    private static void printMessage(String message) {
+        System.out.println(MESSAGE_WALL);
+        System.out.println(message);
+        System.out.println(MESSAGE_WALL);
+    }
+
+    /**
+     * Prints tasks in list format.
+     */
+    private static void printTasks() {
+        StringBuilder messageBuilder = new StringBuilder(MESSAGE_LIST_HEADER);
+        for (int i = 0; i < TASKS.size(); i += 1) {
+            messageBuilder.append("\n");
+            messageBuilder.append(i + 1);
+            messageBuilder.append(": ");
+            messageBuilder.append(TASKS.get(i));
+        }
+
+        printMessage(messageBuilder.toString());
+    }
+
+    /**
+     * Checks if string value is an integer.
+     *
+     * @param string The string to check.
+     * @return true if string can be converted to integer, else false.
+     */
+    private static boolean isStringInteger(String string) {
+        boolean isInt;
+        try {
+            int value = Integer.parseInt(string);
+            isInt = true;
+        } catch (NumberFormatException e) {
+            isInt = false;
+        }
+
+        return isInt;
+    }
+
+    /**
+     * Gets a non-empty user input.
+     *
+     * @return Non-empty user input.
+     */
+    private static String getUserInput() {
+        String input;
+        do {
+            System.out.print(INPUT_PROMPT);
+            input = SCANNER.nextLine();
+        } while (input.trim().isEmpty());
+
+        return input;
+    }
+
+    /**
+     * Splits user input string into command and argument.
+     *
+     * @param input The user input string.
+     * @return String array: [0] - Command, [1] - argument.
+     */
+    private static String[] getCommandAndArgument(String input) {
+        String[] result = input.trim().split("\\s+", 2);
+        return (result.length == 2) ? result : new String[]{result[0], ""};
```

Comment:
Although this is fine, a minor suggestion is to expand out this line to improve code readability as follows:

```
if (result.length != 2) {
    result = new String[]{result[0], ""};
} 
return result;
```